core/src/azure_iot_nx/azure_iot_nx_client.c

```
@@ -104,32 +103,29 @@ static VOID process_direct_method(AZURE_IOT_NX_CONTEXT* nx_context)
     UCHAR* payload;
     USHORT payload_length;
 
-    if ((status = nx_azure_iot_hub_client_direct_method_message_receive(&nx_context->iothub_client,
-             &method_name,
-             &method_name_length,
-             &context,
-             &context_length,
-             &packet,
-             NX_WAIT_FOREVER)))
+    while (nx_azure_iot_hub_client_direct_method_message_receive(&nx_context->iothub_client,
```

If this is not equal to NX_AZURE_IOT_SUCCESS, we will silently return without printing any error code.  We might want to handle this error case instead.

---

This is deliberate because I call this function until it gets an error, so I didn't want to see an error printed everything.

But thinking more, I'll change it to not print an error if it returns no_packet, otherwise I will print.

---

Added error catching